client/app/src/components/dashboard/account-box.controller.js

```diff
@@ -34,7 +34,7 @@
     this.myAccountsCurrencyBalance = () => {
       const market = this.accountCtrl.connectedPeer.market
       const currencyName = this.accountCtrl.currency.name
-      const price = market ? market.price[currencyName] : 0
+      const price = market && market.price ? market.price[currencyName] : 0
```

Comment:
If market or market.price is falsey does the user see 0 for their balance? If that's the case, I'd double check that we throw up a toast error message.

Reply:
I just hit a similar issue to this on my PR. market is undefined until a connection to a peer is established. It resolves itself as soon as that connection happens (which already has toasts related to it) 🤔 I thought about setting to zero initially too. Perhaps we should cache the market data so we at least show something?

Reply:
I like the idea of caching the most recently retrieved value. That still leaves the case where the user doesn't have the cache built yet, but that'd be a big improvement over displaying 0 for returning users.

Reply:
Yeh probably the best way to go!I think I'll do the same on my PR, default to zero. The caching / loading of cache should be done either here or another PR I think. As long as it's done before the next release of the desktop client, no one will notice too much! Perhaps if there's no market data, we show a hyphen (`-`)?

I guess the process would be along the lines of:
- Check cache for market data (load if available)
- Fetch market data from API
- Update live and cache data

Does that sound okay?

Reply:
If the get request for the coinmarket api is faulty, then the rejection function uses the cached value. If there is no cached value, then we return zero. 

I've gone ahead and added a toast error message in that case. Saying we're unable to get market data.

Reply:
I think if we could also cache the data within `updatePeerWithCurrencies` too, then load that as early as possible on client launch by default. The user may not then experience seeing zero at all, once that cache has been populated. You could override the cache assuming the coinmarketcap process is available.

I've commented possible caching changes in this PR.

I home that makes sense?